src/averageAges.js

```diff
@@ -37,7 +44,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => withChildren
+    ? isMother(people, person)
+    : person.sex === 'f'
+  );
+
+  return women.reduce((sum, woman) =>
+    sum + (woman.died - woman.born), 0) / women.length;
+}
+
+function isMother(people, woman) {
+  return people.some(
+    child => child.mother === woman.name
+  );
 }
```

Comment:
Nice one 👍 I just thing that it's possible to corrupt data and add some man name to `mother` property, so will be better to check `sex` too